examples/jsm/controls/TrackballControls.js

```diff
@@ -640,46 +664,45 @@ class TrackballControls extends EventDispatcher {
 
 		}
 
-		function touchmove( event ) {
+		function onTouchMove( event ) {
 
-			if ( scope.enabled === false ) return;
-
-			event.preventDefault();
+			trackPointer( event );
 
-			switch ( event.touches.length ) {
+			switch ( _pointers.length ) {
 
 				case 1:
 					_movePrev.copy( _moveCurr );
-					_moveCurr.copy( getMouseOnCircle( event.touches[ 0 ].pageX, event.touches[ 0 ].pageY ) );
+					_moveCurr.copy( getMouseOnCircle( event.pageX, event.pageY ) );
 					break;
 
 				default: // 2 or more
-					const dx = event.touches[ 0 ].pageX - event.touches[ 1 ].pageX;
-					const dy = event.touches[ 0 ].pageY - event.touches[ 1 ].pageY;
+
+					const position = getSecondPointerPosition( event );
+
+					const dx = event.pageX - position.x;
+					const dy = event.pageY - position.y;
 					_touchZoomDistanceEnd = Math.sqrt( dx * dx + dy * dy );
 
-					const x = ( event.touches[ 0 ].pageX + event.touches[ 1 ].pageX ) / 2;
-					const y = ( event.touches[ 0 ].pageY + event.touches[ 1 ].pageY ) / 2;
+					const x = ( event.pageX + position.x ) / 2;
+					const y = ( event.pageY + position.y ) / 2;
 					_panEnd.copy( getMouseOnScreen( x, y ) );
```

Comment:
What about this?

```js
const pointer1 = _pointers[ 0 ];
const pointer2 = _pointers[ 1 ];

const dx = pointer2.x - pointer1.x;
const dy = pointer2.y - pointer1.y;

_touchZoomDistanceEnd = Math.sqrt( dx * dx + dy * dy );

const x = ( pointer2.x - pointer1.x ) / 2;
const y = ( pointer2.y - pointer1.y ) / 2;
_panEnd.copy( getMouseOnScreen( x, y ) );
```

We shouldn't need `getSecondPointerPosition()` then?

Reply:
`_pointers` holds all active pointers by storing their initial event.  The current position of a pointer is tracked in `_pointerPositions`. So accessing `_pointers` does not work in `onTouchMove()` since the latest event data are required. These data are represented by the passed `event` object and the tracked data of the second pointer.

I have found it easier to manage pointers in two data structures (`_pointers` and `_pointerPositions`).

Reply:
Hmm, okay! We can do more clean up later if necessary 🤓